File: javascript/word-count/word-count.js
var words = function(input) {
  var inputArr = input.replace(/[\t\n]/g, ' ').split(' ')
  var result = {}
  for (var i = 0, len = inputArr.length; i < len; i++) {
    if (inputArr[i] in result) {
      result[inputArr[i]] += 1
    } else {
      result[inputArr[i]] = 1
    }
  }
  return result
};

module.exports = words;

File: elixir/word-count/word_count.exs
defmodule Words do
  def count(sentence) do
    list = list_from_sentence(sentence)
    unique_list = Enum.uniq(list)

    map = Enum.into(unique_list, %{},
      fn(word) ->
        { word, count_in_list(list, word) }
      end
    )
    Map.delete(map, "")
  end

  def list_from_sentence(sentence) do
    sentence
    |> String.replace(~r/!|"|\#|\$|%|&|'|\(|\)|\*|\+|,|\.|\/|:|;|<|=|>|\?|@|\[|\\|]|\^|_|`|\{|\||}|~/, " ")
    |> String.downcase
    |> String.split(" ")
  end

  def count_in_list(list, word) do
    Enum.count(list,
      fn(item) ->
        item == word
      end
    )
  end
end
